.github/workflows/clippy_bors.yml

```diff
@@ -0,0 +1,168 @@
+name: Test Clippy (bors)
+
+on:
+  push:
+    branches: [auto, try]
+    # Don't run Clippy tests, when only textfiles were modified
+    paths-ignore:
+    - 'COPYRIGHT'
+    - 'LICENSE-*'
+    - '**.md'
+    - '**.txt'
+  pull_request:  # TODO: Remove this
+    # Don't run Clippy tests, when only textfiles were modified
+    paths-ignore:
+    - 'COPYRIGHT'
+    - 'LICENSE-*'
+    - '**.md'
+    - '**.txt'
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_TARGET_DIR: '${{ github.workspace }}/target'
+  GHA_CI: 1
+
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check Changelog
+      run: |
+        PR=$(echo "${{ github.ref }}" | grep -o "[0-9]*")
+        output=$(curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -s "https://api.github.com/repos/rust-lang/rust-clippy/pulls/$PR" | \
+          python -c "import sys, json; print(json.load(sys.stdin)['body'])" | \
+          grep "^changelog: " | \
+          sed "s/changelog: //g")
+        if [[ -z "$output" ]]; then
+          echo "ERROR: PR body must contain 'changelog: ...'"
+          exit 1
+        elif [[ "$output" = "none" ]]; then
+          echo "WARNING: changelog is 'none'"
+        fi
+  base:
+    needs: changelog
+    strategy:
+      fail-fast: false  # TODO: remove later
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        host: [x86_64-unknown-linux-gnu, i686-unknown-linux-gnu, x86_64-apple-darwin, x86_64-pc-windows-msvc]
+        exclude:
+        - os: ubuntu-latest
+          host: x86_64-apple-darwin
+        - os: ubuntu-latest
+          host: x86_64-pc-windows-msvc
+        - os: macos-latest
+          host: x86_64-unknown-linux-gnu
+        - os: macos-latest
+          host: i686-unknown-linux-gnu
+        - os: macos-latest
+          host: x86_64-pc-windows-msvc
+        - os: windows-latest
+          host: x86_64-unknown-linux-gnu
+        - os: windows-latest
+          host: i686-unknown-linux-gnu
+        - os: windows-latest
+          host: x86_64-apple-darwin
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Install dependencies (Linux-i686)
+      run: |
+        sudo dpkg --add-architecture i386
+        sudo apt-get update
+        sudo apt-get install gcc-multilib libssl-dev:i386
+      if: matrix.host == 'i686-unknown-linux-gnu'
+    - name: rust-toolchain
+      uses: actions-rs/toolchain@v1.0.3
+      with:
+        toolchain: nightly
+        target: ${{ matrix.host }}
+        profile: minimal
+    - name: Cache cargo dir
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo
+        key: ${{ runner.os }}-cargo
+    - name: Checkout
+      uses: actions/checkout@v2.0.0
+    - name: Master Toolchain Setup
+      run: bash setup-toolchain.sh
+      env:
+        HOST_TOOLCHAIN: ${{ matrix.host }}
+      shell: bash
+
+    - name: Set SYSROOT path (Linux)
+      if: runner.os == 'Linux'
+      run: |
+        SYSROOT=$(rustc --print sysroot)
```

Comment:
I think you will need `export` here to make `SYSROOT` persist between multiple shells.

Reply:
We don't want to persist sysroot. This is just c&p from the Travis config. The SYSROOT var is pretty useless and we could just directly write $(rustc...) below.

But I should check the {LD_LIBRARY_,}PATH env vars again

Reply:
GHA has [`export RUSTUP_HOME=/usr/share/rust/.rustup`](https://github.com/actions/virtual-environments/blob/0b82f907e8893065cc512b440a4812a73a6bebcc/images/linux/scripts/installers/rust.sh#L12) so Clippy driver should take use during compilation: https://github.com/rust-lang/rust-clippy/blob/f78cc075099f608787d50e151b4b6f529a0629c1/src/driver.rs#L310-L311

It's going to be hard to tell without debugging the driver.

Reply:
Weird... None of these vars are set in GHA: https://github.com/rust-lang/rust-clippy/runs/402834991

But `rustc --print sysroot` is `C:/Rust/.rustup`/`usr/share/rust/.rustup` :thinking:

Reply:
rustc 1.39.0 is preinstalled on [Ubuntu](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#ubuntu-1804-lts) and [Windows](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#rust-64-bit-1). Therefore rustup is cemented into `/usr/share/rust/.rustup`. 

The weird thing is, that the clippy-driver should use the sysroot provided by `rustc --print sysroot` if non of the env vars are provided.

Reply:
On the other hand, `rustup show`  prints `$HOME/.rustup` as sysroot, but then goes on and installs everything to `/usr/share/rust/.rustup`.

Reply:
I have an idea how to explain it.
Rustup probably defines `RUSTUP_HOME` when proxy binaries (`/usr/share/rust/.cargo/bin/{cargo,rustc}`) are called. But it points to the wrong location (`$HOME/.rustup`) and Clippy hardcodes this location via `option_env!("RUSTUP_HOME")` during the compilation. This has preference over `rustc --print sysroot` and `rustc` doesn't care about `RUSTUP_HOME`. That would explain the discrepancy.

To prove it you could call cargo directly (`/usr/share/rust/.rustup/toolchains/master/bin/cargo`) instead of the `cargo`.

Reply:
@mati865 yes, just discovered this myself https://github.com/rust-lang/rust-clippy/pull/5071#issuecomment-577188796 :smile: